diff --git a/third-party/CMakeLists.txt b/third-party/CMakeLists.txt
index 7456d749f3..6549a51565 100644
--- a/third-party/CMakeLists.txt
+++ b/third-party/CMakeLists.txt
@@ -30,6 +30,7 @@ ExternalProject_Add(
   flatbuffers_external_project
   PREFIX ${CMAKE_CURRENT_BINARY_DIR}/flatbuffers_external_project
   SOURCE_DIR ${PROJECT_SOURCE_DIR}/third-party/flatbuffers
+  CMAKE_GENERATOR_PLATFORM "x64"
   CMAKE_ARGS -DFLATBUFFERS_BUILD_FLATC=ON
              -DFLATBUFFERS_INSTALL=ON
              -DFLATBUFFERS_BUILD_FLATHASH=OFF
@@ -42,7 +43,7 @@ ExternalProject_Add(
              # If building for iOS, "unset" these variables to rely on the host (macOS) defaults.
              $<$<AND:$<BOOL:${APPLE}>,$<BOOL:$<FILTER:${PLATFORM},EXCLUDE,^MAC>>>:-DCMAKE_OSX_SYSROOT=>
              -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET}
-  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/flatc
+  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/flatc$<$<BOOL:WIN32>:.exe>
   ${_executorch_external_project_additional_args}
 )
 ExternalProject_Get_Property(flatbuffers_external_project INSTALL_DIR)
diff --git a/tools/cmake/Codegen.cmake b/tools/cmake/Codegen.cmake
index 93331c7ed8..bb3b46759d 100644
--- a/tools/cmake/Codegen.cmake
+++ b/tools/cmake/Codegen.cmake
@@ -122,7 +122,8 @@ function(generate_bindings_for_kernels)
   execute_process(
     COMMAND
       "${PYTHON_EXECUTABLE}" -c
-      "import torchgen;import os; print(os.path.dirname(torchgen.__file__))"
+      "import importlib.util, os; s=importlib.util.find_spec('torchgen'); \
+         print(s.submodule_search_locations[0] if s and s.submodule_search_locations else os.path.dirname(s.origin))"
     OUTPUT_VARIABLE torchgen-out
     ERROR_VARIABLE torchgen-out-error
     RESULT_VARIABLE torchgen-result
