diff --git a/runtime/platform/compat_unistd.h b/runtime/platform/compat_unistd.h
index c8bc486670..bd1c30f19f 100644
--- a/runtime/platform/compat_unistd.h
+++ b/runtime/platform/compat_unistd.h
@@ -12,12 +12,7 @@
  */
 #pragma once
 
-#if defined(_WIN32) && !defined(_WIN64)
-#error \
-    "You're trying to build ExecuTorch with a too old version of Windows. We need Windows 64-bit."
-#endif
-
-#if !defined(_WIN64)
+#if !defined(_WIN64) && !defined(_WIN32)
 #include <unistd.h>
 #else
 #include <io.h>
