diff --git a/tools/cmake/Utils.cmake b/tools/cmake/Utils.cmake
index 9fbab17728..6179fd3994 100644
--- a/tools/cmake/Utils.cmake
+++ b/tools/cmake/Utils.cmake
@@ -88,6 +88,8 @@ function(extract_sources sources_file)
     if(ANDROID_ABI)
       if("${ANDROID_ABI}" STREQUAL "arm64-v8a")
         set(target_platforms_arg "--target-platforms=shim_et//:android-arm64")
+      elseif("${ANDROID_ABI}" STREQUAL "armeabi-v7a")
+        set(target_platforms_arg "--target-platforms=shim_et//:android-arm32")
       elseif("${ANDROID_ABI}" STREQUAL "x86_64")
         set(target_platforms_arg "--target-platforms=shim_et//:android-x86_64")
       else()
