diff --git a/install_requirements.sh b/install_requirements.sh
index ef156cd020..0e20235b28 100755
--- a/install_requirements.sh
+++ b/install_requirements.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 # Copyright (c) Meta Platforms, Inc. and affiliates.
 # All rights reserved.
 #
diff --git a/run_python_script.sh b/run_python_script.sh
index 4f9a74ec36..00e5912921 100755
--- a/run_python_script.sh
+++ b/run_python_script.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 # Copyright (c) Meta Platforms, Inc. and affiliates.
 # All rights reserved.
 #
